llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -242,25 +242,48 @@ ModulePass *llvm::createYkControlPointPass(uint64_t controlPointCount) {
 // Returns true iff the function contains a (patched or unpatched) control
 // point.
 bool llvm::containsControlPoint(llvm::Function &F) {
+  static llvm::DenseMap<llvm::Function*, bool> cache;
+
+  auto it = cache.find(&F);
+  if (it != cache.end()) {
+    return it->second;
+  }
+
+  // Slow path
+  static const StringRef CP_PP_NAME_REF(CP_PPNAME);
```

Comment:
I'm always a bit nervous about storing stringrefs.

I *think* this is OK in this case though?

> This class does not own the string data, it is expected to be used in situations where the character data resides in some other buffer, whose lifetime extends past that of the [StringRef](https://llvm.org/doxygen/classllvm_1_1StringRef.html). For this reason, it is not in general safe to store a StringRef.

https://llvm.org/doxygen/classllvm_1_1StringRef.html#details

If there's any doubt, I think we should just store static `char *` and construct stringrefs as needed.

Reply:
Agreed, this is not safe in the general case, but because the names of the intrinsics live in the data segment the lifetime of the `StringRef`s will always be valid. This will give us faster comparisons.